### PR TITLE
docs(tuning): updates to log compaction description

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Updates the overview of log cleanup policies in the broker tuning documentation.  

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by PaulRMellor** [comment]
Small change, but I would like to request that the diagrams that accompany the section also be reviewed. I have received feedback that they may not accurately show the compaction and combined compaction and delete processes.

**Review by scholzj — APPROVED** [review]

File: documentation/modules/managing/con-broker-config-properties.adoc [hunk 1]
```diff
@@ -296,15 +296,16 @@ The frequency of cleanup should be often enough to manage the disk space but not
 
 .Retaining the most recent messages using compact policy
 
-The compact policy guarantees that the most recent message for each message key is retained. 
+When you enable log compaction for a topic by setting `cleanup.policy=compact`, Kafka uses the log cleaner as a background process to perform the compaction.
```
Comment (fvaleri) [inline documentation/modules/managing/con-broker-config-properties.adoc:299]:
```suggestion
When you enable log compaction for a topic by setting `cleanup.policy=compact`, Kafka uses the log cleaner as a background thread to perform the compaction.
It is one thread by default, but you can add more using the `log.cleaner.threads` configuration.
```

File: documentation/modules/managing/con-broker-config-properties.adoc [hunk 2]
```diff
@@ -296,15 +296,16 @@ The frequency of cleanup should be often enough to manage the disk space but not
...
 If a cleanup policy is set for log compaction, the _head_ of the log operates as a standard Kafka log, with writes for new messages appended in order. 
 In the _tail_ of a compacted log, where the log cleaner operates, records are deleted if another record with the same key occurs later in the log. 
 Messages with null values are also deleted. 
 To use compaction, you must have keys to identify related messages because Kafka guarantees that the latest messages for each key will be retained, but it does not guarantee that the whole compacted log will not contain duplicates.
 
-.Log Showing Key Value Writes with Offset Positions Before Compaction
+.Log showing key value writes with offset positions before compaction
 image::tuning/broker-tuning-compaction-before.png[Image of compaction showing key value writes]
 
 Using keys to identify messages, Kafka compaction keeps the latest message (with the highest offset) for a specific message key, eventually discarding earlier messages that have the same key.
```
Comment (fvaleri) [inline documentation/modules/managing/con-broker-config-properties.adoc:311]:
```suggestion
Using keys to identify messages, Kafka compaction keeps the latest message (with the highest offset) that is present in the log tail for a specific message key, eventually discarding earlier messages that have the same key. 
```

File: documentation/modules/managing/con-broker-config-properties.adoc [hunk 3]
```diff
@@ -314,32 +315,23 @@ Records retain their original offsets even when surrounding records get deleted.
...
+---- 
 The deleted data retention period gives time to notice the data is gone before it is irretrievably deleted.
 
-To delete all messages related to a specific key, a producer can send a _tombstone_ message.
-A _tombstone_ has a null value and acts as a marker to tell a consumer the value is deleted.
-After compaction, only the tombstone is retained, which must be for a long enough period for the consumer to know that the message is deleted.
-When older messages are deleted, having no value, the tombstone key is also deleted from the partition.
+To delete all messages related to a specific key, a producer can send a _tombstone_ message. 
+A tombstone has a null value and acts as a marker to inform consumers that the corresponding message for that key has been deleted. 
+After compaction, only the tombstone marker is retained.
```
Comment (fvaleri) [inline documentation/modules/managing/con-broker-config-properties.adoc:333]:
```suggestion
After some time, only the tombstone marker is retained. 
```

File: documentation/modules/managing/con-broker-config-properties.adoc [hunk 4]
```diff
@@ -314,32 +315,23 @@ Records retain their original offsets even when surrounding records get deleted.
...
 The deleted data retention period gives time to notice the data is gone before it is irretrievably deleted.
 
-To delete all messages related to a specific key, a producer can send a _tombstone_ message.
-A _tombstone_ has a null value and acts as a marker to tell a consumer the value is deleted.
-After compaction, only the tombstone is retained, which must be for a long enough period for the consumer to know that the message is deleted.
-When older messages are deleted, having no value, the tombstone key is also deleted from the partition.
+To delete all messages related to a specific key, a producer can send a _tombstone_ message. 
+A tombstone has a null value and acts as a marker to inform consumers that the corresponding message for that key has been deleted. 
+After compaction, only the tombstone marker is retained. 
+The marker is retained for a duration specified by `log.cleaner.delete.retention.ms` to allow consumers enough time to recognize the deletion.
```
Comment (fvaleri) [inline documentation/modules/managing/con-broker-config-properties.adoc:334]:
```suggestion
Assuming new messages continue to come in, the marker is retained for a duration specified by `log.cleaner.delete.retention.ms` to allow consumers enough time to recognize the deletion.
```

**Review by fvaleri — COMMENTED** [review]
Left some suggestions, but diagrams LGTM.

Records before the retention point (0 and 1) are deleted in the after compaction diagram. I think this may create some confusion.

File: documentation/modules/managing/con-broker-config-properties.adoc [hunk 5]
```diff
@@ -350,6 +342,21 @@ log.cleaner.backoff.ms=15000
 # ...
 ----
 
+.Using combined compact and delete policies 
+
+If you choose only a compact policy, your log can still become arbitrarily large. 
+In such cases, you can set the cleanup policy for a topic to compact and delete logs.
+Kafka first applies log compaction, removing older versions of records and retaining only the latest version of each key.
```
Comment (mimaison) [inline documentation/modules/managing/con-broker-config-properties.adoc:349]: It's not really applying compaction "first". Both cleanup policies are applied in parallel. In practice and with default settings, compaction happens more regularly than deletion so it appears to happen "first" but you could have configs where deletion appears to happen "first".

**Review by ppatierno — APPROVED** [review]

**Review by fvaleri — APPROVED** [review]

**Review by mimaison — APPROVED** [review]
LGTM

**Pull request merged**